Dockerfile

```diff
@@ -1,11 +1,57 @@
+# syntax=docker/dockerfile:1.2
+
 #---------------------------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 #---------------------------------------------------------------------------------------------
 
 ARG PYTHON_VERSION="3.10"
 
-FROM python:${PYTHON_VERSION}-alpine
+FROM python:${PYTHON_VERSION}-alpine AS base
+
+# We don't use openssl (3.0) for now. We only install it so that users can use it.
+# libintl and icu-libs - required by azure devops artifact (az extension add --name azure-devops)
+RUN apk add --no-cache \
+    bash bash-completion openssh-keygen ca-certificates openssl \
+    curl jq git perl zip \
+    libintl icu-libs libc6-compat \
+ && update-ca-certificates
+
+#---------------------------------------------------------------------------------------------
+
+FROM base AS builder
+
+# bash gcc make openssl-dev libffi-dev musl-dev - dependencies required for CLI
+
+RUN apk add --no-cache --virtual .build-deps gcc make openssl-dev libffi-dev musl-dev linux-headers
+
+WORKDIR azure-cli
+COPY . /azure-cli
+
+# 1. Build packages and store in tmp dir
+# 2. Install the cli and the other command modules that weren't included
+# 3. Run trim_sdk.py to clean up un-used py files: https://github.com/Azure/azure-cli/pull/25801
+RUN ./scripts/install_full.sh
+RUN python3 ./scripts/trim_sdk.py
+
+# Normalize line endings.
+RUN dos2unix /usr/local/bin/az /usr/local/bin/az.completion.sh
```

Comment:
I wonder why we have to run dos2unix.  Is that a CI issue?

Reply:
This seems like a debt to me. However, now it's isolated to the build step so I'm cool with it. There is however this cost - causing confusion and questions about intent and why.

Reply:
The repo should enforce a unix line ending standard so that this isn't needed.